app/src/main/java/be/scri/ui/models/ScribeItem.kt

```diff
@@ -0,0 +1,41 @@
+package be.scri.ui.models
+
+sealed class ScribeItem(
+    open val title: String,
+    open val desc: String?,
+) {
+
+    data class ClickableItem(
+        override val title: String,
+        override val desc: String? = null,
+        val action: () -> Unit
+    ) : ScribeItem(title, desc)
+
+    data class SwitchItem(
+        override val title: String,
+        override val desc: String,
+        val state: Boolean,
+        val onToggle: (Boolean) -> Unit
+    ) : ScribeItem(title, desc)
+
+    data class ExternalLinkItem(
+        override val title: String,
+        override val desc: String,
+        val url: String,
+        val onClick: (String) -> Unit
+    ) : ScribeItem(title, desc)
+
+    data class CustomItem(
+        override val title: String,
+        override val desc: String,
+        val customAction: (Any?) -> Unit
+    ) : ScribeItem(title, desc)
+}
+
+
+enum class ItemType {
```

Comment:
This is currently not being used right and could be removed or the codebase could be modified to use them.

Reply:
The enum class is no more of use and the other two data classes can be used in the future when we migrate the About page to compose